index/index.go

```diff
@@ -494,8 +513,9 @@ func (w *Writer) WritePostings(name, value string, it Postings) error {
 		return err
 	}
 
-	w.postings = append(w.postings, hashEntry{
-		keys:   []string{name, value},
+	w.postings = append(w.postings, postingsHashEntry{
```

Comment:
I'm just surprised that this small change reduces allocs so much.

Reply:
Quite interesting indeed, but keep in mind that slice header is like 16 bytes overall, so in some cases might be bigger than `name` or `key` (: